doc/yaml_configuration.md

```diff
@@ -1045,3 +1045,25 @@ a yaml configuration file.
 
 (The British English spelling (colour) is also accepted. In yaml configuration
 files, the American spelling is the alternative that has priority.)
+
+### hide-source-directories
+
+Stack will use the `-fhide-source-paths` option by default for GHC >= 8.2, unless this
+option is set to `false`. 
+
+When enabled:
+
+```
+...
+[1 of 2] Compiling Lib
+[2 of 2] Compiling Paths_test_pr
+...
+```
+
+When disabled:
+
+```
+...
+[1 of 2] Compiling Lib              ( src/Lib.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/Lib.o )
+...
+```
```

Comment:
Please also include:

1. A small `yaml` snippet showing usage, like we have above for others
2. A comment saying `Since 2.0`

Reply:
Ok, thank you!